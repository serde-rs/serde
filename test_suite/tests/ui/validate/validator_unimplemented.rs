use serde_derive::Deserialize;

#[derive(Deserialize)]
#[serde(validator)]
struct ValidatorStruct {
    mail: String,
}

fn main() {}
