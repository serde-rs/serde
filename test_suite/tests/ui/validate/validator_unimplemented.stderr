error[E0277]: the trait bound `ValidatorStruct: Validate` is not satisfied
 --> tests/ui/validate/validator_unimplemented.rs:3:10
  |
3 | #[derive(Deserialize)]
  |          ^^^^^^^^^^^ the trait `Validate` is not implemented for `ValidatorStruct`
  |
  = help: the following other types implement trait `Validate`:
            &BTreeMap<K, V>
            &HashMap<K, V, S>
            &T
            BTreeSet<T>
            BinaryHeap<T>
            HashSet<T>
            LinkedList<T>
            Vec<T>
          and $N others
  = note: this error originates in the derive macro `Deserialize` (in Nightly builds, run with -Z macro-backtrace for more info)
